x-pack/test/security_api_integration/tests/anonymous/login.ts

```diff
@@ -167,6 +167,7 @@ export default function ({ getService }: FtrProviderContext) {
 
         expect(apiResponse.body.statusCode).to.be(401);
         expect(apiResponse.body.error).to.be('Unauthorized');
+        // TODO: Fix type failure introduced by elasticsearch-js 8.6.0-canary.3
         expect(apiResponse.body.message).to.include.string(
           '[security_exception] Reason: unable to authenticate user [dummy_hacker] for REST request [/_security/_authenticate]'
```

Comment:
@elastic/kibana-security This test fails with:
```
expected '[security_exception\n\tRoot causes:\n\t\tsecurity_exception: unable to authenticate user [dummy_hacker] for REST request [/_security/_authenticate]]: unable to authenticate user [dummy_hacker] for REST request [/_security/_authenticate]' to contain '[security_exception] Reason: unable to authenticate user [dummy_hacker] for REST request [/_security/_authenticate]'
```

I'm not sure why this behaviour changed with the upgrade. Perhaps it's fine to just update the expected response?

Reply:
>I'm not sure why this behaviour changed with the upgrade. 

Not a blocker, but can we ping ES JS client maintainers to clarify this change in error message/shape? Just to get a better understanding and make sure it's not a sign of something else that we should be worried about.

Reply:
https://github.com/elastic/elastic-transport-js/pull/52 introduced a change to the error.message property. It seems like depending on the `error.root_cause` might be more stable.

/cc @sethmlarson

Reply:
Awesome, thank you for digging this up @rudolf !